pymeasure/adapters/prologix.py

```diff
@@ -144,42 +165,52 @@ def write_binary_values(self, command, values, **kwargs):
         """
         if self.address is not None:
             address_command = "++addr %d\n" % self.address
-            self.connection.write(address_command.encode())
-        super().write_binary_values(command, values, **kwargs)
-        self.connection.write(b'\n')
+            self.write(address_command)
+        super().write_binary_values(command, values, "\n", **kwargs)
 
-    def read(self):
-        """ Reads the response of the instrument until timeout
+    def _read(self, **kwargs):
+        """Read up to (excluding) `read_termination` or the whole read buffer.
 
-        :returns: String ASCII response of the instrument
+        :param kwargs: Keyword arguments for the connection itself.
+        :returns str: ASCII response of the instrument (excluding read_termination).
         """
         self.write("++read eoi")
-        return b"\n".join(self.connection.readlines()).decode()
+        return super()._read()
```

Comment:
Since long time, I  wanted to get rid of that `readlines` which added always a timeout and also read several lines at once.
I like your changes and I think solves both problems.

Reply:
Thanks.